### DISCUSSION TITLE
Linux Automated Task Runner

### DISCUSSION THREAD

**Discussion by PirvuCatalin** [amidaware/tacticalrmm#1692] [General]
Hi,
Seems that it wasn't quite clear for me that Automated Task Runner (Scheduled Tasks) was only working with Windows agents.

Now that I know that, I'm willing to work on making this work with linux agents as well, but I'd use some guidelines for where to start. 
A link to the code that deals with tasks creation would work as well.

The use case is quite simple, I want to turn on and off devices at specific times during the day. In the future, maybe also create things like monthly updating some linux machines.

For now the alternative is to use https://github.com/ryanblenis/MeshCentral-ScriptTask for scheduling tasks in MeshCentral, but it doesn't have the flexibility of Tactical dashboard.
If you know about something in progress which I can check, that can also work.

Thank you!

**Comment by wh1te909** [comment]
We are working on this but still experimenting with implementation so code is not public yet and will not be until we have finalized an implementation. In the meantime this is pretty simple to implement yourself using trmm's api. just write a python script or something that calls the trmm api to run the trmm script or command on the agent(s) you wish to run, then throw this python script in crontab somewhere. you have now implemented automated tasks for linux agents.

**Comment by silversword411** [comment]
Please note: your definition and wh1te909s definition of "Pretty simple" may not completely match. 🙂

**Reply by wh1te909** [reply]
lol remind me to get an example of this on the docs. it really is trivial xD

**Comment by bbrendon** [comment]
The hardest part is getting the script id :)
Or was that added to the gui? Heh

**Comment by Nerdy-Technician** [comment]
Good Afternoon Just seeing if you have any update regarding this request for the gui ?

**Reply by dinger1986** [reply]
https://docs.tacticalrmm.com/roadmap/

As you can see running a little behind but next in line

**Reply by Nerdy-Technician** [reply]
Yes I have been tracking the roadmap just kind of looking for an update on how its going i.e progress made but thanks for your message

**Comment by janstalhut** [comment]
I have seen the roadmap that refers to this issue. I am very pleased that the Tasks function is coming to Linux systems, as I am currently missing this option a lot. In this context, I was wondering whether this would also include support for Collector Tasks. I think these are particularly important for gathering individual details about individual systems.

**Reply by wh1te909** [reply]
yes it will include collector tasks

**Comment by Nerdy-Technician** [comment]
Good Afternoon,

Just following on my message from a few months ago how is the progress going with Linux tasks into TRMM, 

Hope you are well and love your product!!

Many thansk

**Reply by wh1te909** [reply]
the feature is complete, it's entering the testing phase so should be a couple more weeks

**Reply by AlvaroLCT** [reply]
great to hear that, we are a new sponsor and we want to continue sponsoring for many time

**Comment by jamie-dit** [comment]
Hi @wh1te909 

How are we looking for Linux Tasks?

Did you need any testers?

Cheers

**Reply by wh1te909** [reply]
testing is complete, thanks. just need to record video and update docs and we're good for release, aiming for next week

**Comment by wh1te909** [comment]
feature is now release in version 1.0.0
